spark/v3.1/spark/src/main/java/org/apache/iceberg/spark/SparkConfParser.java

```diff
@@ -132,6 +132,10 @@ public String parse() {
       Preconditions.checkArgument(defaultValue != null, "Default value cannot be null");
       return parse(Function.identity(), defaultValue);
     }
+
+    public String parseOptional() {
```

Comment:
Can you please make Spark changes to the 3.3 branch only and then backport once the initial commit is done?